utilities/rpp-unittests/OCL/Single_ocl.cpp

```diff
@@ -0,0 +1,595 @@
+	RppiSize *dstSize = (RppiSize *)calloc(noOfImages, sizeof(RppiSize));
+	//const int images = noOfImages;
+	char imageNames[images][1000];
+
+	unsigned long long count1 = 0;
+
+	DIR *dr1 = opendir(src);
+	while ((de = readdir(dr1)) != NULL)
+	{
+		if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+			continue;
+		strcpy(imageNames[count1], de->d_name);
+		char temp[1000];
+		strcpy(temp, src1);
+		strcat(temp, imageNames[count1]);
+		if (ip_channel == 3)
+		{
+			image = imread(temp, 1);
+		}
+		else
+		{
+			image = imread(temp, 0);
+		}
+		srcSize[count1].height = image.rows;
+		srcSize[count1].width = image.cols;
+		ioBufferSize += (unsigned long long)srcSize[count1].height * (unsigned long long)srcSize[count1].width * (unsigned long long)ip_channel;
+
+		count1++;
+		break;
+	}
+	closedir(dr1);
+
+	Rpp8u *input = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+	Rpp8u *input_second = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+	Rpp8u *output = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+
+	/* Read the input image */
+	DIR *dr2 = opendir(src);
+	DIR *dr2_second = opendir(src_second);
+	unsigned long long count2 = 0;
+	i = 0;
+	while ((de = readdir(dr2)) != NULL)
+	{
+		if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+			continue;
+		char temp[1000];
+		strcpy(temp, src1);
+		strcat(temp, de->d_name);
+		char temp_second[1000];
+		strcpy(temp_second, src1_second);
+		strcat(temp_second, de->d_name);
+		if (ip_channel == 3)
+		{
+			image = imread(temp, 1);
+			image_second = imread(temp_second, 1);
+		}
+		else
+		{
+			image = imread(temp, 0);
+			image_second = imread(temp_second, 0);
+		}
+		Rpp8u *ip_image = image.data;
+		Rpp8u *ip_image_second = image_second.data;
+		for (j = 0; j < srcSize[i].height * srcSize[i].width * ip_channel; j++)
+		{
+			input[count2] = ip_image[j];
+			input_second[count2] = ip_image_second[j];
+			count2++;
+		}
+		i++;
+		break;
+	}
+
+	closedir(dr2);
+	cl_mem d_input, d_input_second, d_output;
+	cl_platform_id platform_id;
+	cl_device_id device_id;
+	cl_context theContext;
+	cl_command_queue theQueue;
+	cl_int err;
+	err = clGetPlatformIDs(1, &platform_id, NULL);
+	err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_GPU, 1, &device_id, NULL);
+	theContext = clCreateContext(0, 1, &device_id, NULL, NULL, &err);
+	theQueue = clCreateCommandQueue(theContext, device_id, 0, &err);
+	d_input = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+	d_input_second = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+	d_output = clCreateBuffer(theContext, CL_MEM_READ_ONLY, ioBufferSize * sizeof(Rpp8u), NULL, NULL);
+	err = clEnqueueWriteBuffer(theQueue, d_input, CL_TRUE, 0, ioBufferSize * sizeof(Rpp8u), input, 0, NULL, NULL);
+	err = clEnqueueWriteBuffer(theQueue, d_input_second, CL_TRUE, 0, ioBufferSize * sizeof(Rpp8u), input_second, 0, NULL, NULL);
```

Comment:
Codacy found an issue: [Variable 'err' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)